omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AdministrationLinksResource2_0.java

```diff
@@ -0,0 +1,160 @@
+import org.openmrs.module.web.extension.AdministrationSectionExt;
+import org.openmrs.module.webservices.helper.ModuleFactoryWrapper;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingReadableResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Resource(name = RestConstants.VERSION_1 + "/administrationlinks", supportedClass = AdministrationSectionExt.class,
+		supportedOpenmrsVersions = { "2.0.*", "2.1.*", "2.2.*", "2.3.*", "2.4.*", "2.5.*" })
+public class AdministrationLinksResource2_0 extends BaseDelegatingReadableResource<AdministrationSectionExt> {
+
+	private static final String ADMIN_LIST_POINT_ID = "org.openmrs.admin.list";
+
+	private static final String MODULE_TITLE = "title";
+
+	private static final String LINKS = "administrationLinks";
+
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
+
+	@Override
+	public AdministrationSectionExt newDelegate() {
+		return null;
+	}
+
+	@Override
+	public AdministrationSectionExt getByUniqueId(String moduleId) {
+		// Assumes that moduleId is an id of a module that user wants to get admin links for.
+		// AdministrationLinksResource has no id by itself as it's not an OpenMRS data object.
+
+		AdministrationSectionExt administrationLinks = getAdministrationLinksForModule(moduleId);
+		if (administrationLinks == null) {
+			throw new ObjectNotFoundException(
+					"Module with id: " + moduleId + " doesn't have any administration links registered.");
+		}
+
+		return administrationLinks;
+	}
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addProperty(MODULE_TITLE);
```

Comment:
Why use constants for these and yet uuid and display are not in constants?

Reply:
That would cause inconsistency since every other resource doesn't use constants for uuid and display. However, I could do so for that resource.